crates/ruff_linter/src/rules/ruff/rules/split_of_static_string.rs

```diff
@@ -0,0 +1,166 @@
+    // > a None separator returns [].
+    // https://docs.python.org/3/library/stdtypes.html#str.split
+    if max_split == 0 {
+        let list_items: Vec<&str> = str_value.split_whitespace().collect();
+        Some(construct_replacement(&list_items))
+    } else {
+        // Autofix for maxsplit without separator not yet implemented
+        None
+    }
+}
+
+fn split_sep(str_value: &str, sep_value: &str, max_split: usize, direction_left: bool) -> Expr {
+    let list_items: Vec<&str> = if direction_left && max_split > 0 {
+        str_value.splitn(max_split + 1, sep_value).collect()
+    } else if !direction_left && max_split > 0 {
+        str_value.rsplitn(max_split + 1, sep_value).collect()
+    } else if direction_left && max_split == 0 {
+        str_value.split(sep_value).collect()
+    } else {
+        str_value.rsplit(sep_value).collect()
+    };
+    construct_replacement(&list_items)
+}
+
+/// RUF035
+pub(crate) fn split_of_static_string(
+    checker: &mut Checker,
+    attr: &str,
+    call: &ExprCall,
+    str_value: &str,
+) {
+    let ExprCall { arguments, .. } = call;
+
+    let sep_arg = arguments.find_argument("sep", 0);
+    let maxsplit_arg = arguments.find_argument("maxsplit", 1);
+
+    // `split` vs `rsplit`
+    let direction_left = attr == "split";
+
+    let maxsplit_value = if let Some(maxsplit) = maxsplit_arg {
+        match maxsplit {
+            Expr::NumberLiteral(maxsplit_val) => {
+                if let Some(int_value) = maxsplit_val.value.as_int() {
+                    if let Some(usize_value) = int_value.as_usize() {
+                        usize_value
+                    } else {
+                        return;
+                    }
+                } else {
+                    return;
+                }
+            }
+            // Ignore when `maxsplit` is not a numeric value
+            _ => {
+                return;
+            }
+        }
+    } else {
+        0
+    };
+
+    let split_replacement = if let Some(sep) = sep_arg {
+        match sep {
+            Expr::NoneLiteral(_) => split_default(str_value, maxsplit_value),
+            Expr::StringLiteral(sep_value) => {
+                let sep_value_str = sep_value.value.to_str();
+                Some(split_sep(
+                    str_value,
+                    sep_value_str,
+                    maxsplit_value,
+                    direction_left,
+                ))
+            }
+            // Ignore names until type inference is available
+            _ => {
+                return;
+            }
+        }
+    } else {
+        split_default(str_value, maxsplit_value)
+    };
+
+    let mut diagnostic = Diagnostic::new(SplitOfStaticString, call.range());
+    if let Some(ref replacement_expr) = split_replacement {
+        // Construct replacement list
+        let replacement = checker.generator().expr(replacement_expr);
+        diagnostic.set_fix(Fix::unsafe_edit(Edit::range_replacement(
```

Comment:
What's your reasoning for making this an unsafe edit?

Reply:
I had not yet exhausted all edge cases (comments, implicit string concatenation, etc.), so I marked it unsafe to be on the _safe_ side. With the new test cases we can probably mark them as safe.